op-node/rollup/driver/state.go

```diff
@@ -358,11 +364,22 @@ func (s *state) eventLoop() {
 				s.idleDerivation = true
 				s.metrics.SetDerivationIdle(true)
 				continue
-			} else if err != nil {
+			} else if err != nil && errors.Is(err, derive.ErrReset) {
 				// If the pipeline corrupts, e.g. due to a reorg, simply reset it
 				s.log.Warn("Derivation pipeline is reset", "err", err)
 				s.derivation.Reset()
 				s.metrics.RecordPipelineReset()
+			} else if err != nil && errors.Is(err, derive.ErrTemporary) {
+				s.log.Warn("Derivation process temporary error", "err", err)
+				// If there's a temporary error, retry with backoff
+				err := backoff.Do(10, bOff, func() error {
+					reqStep()
+					return nil
+				})
+				if err != nil {
+					s.log.Warn("Derivation retry with backoff failed permanently", "err", err)
+				}
```

Comment:
I don't think this works. If the backoff function does not return an error, the backoff time will never increase. I'd like it to request the next step, and update some local state in this function above the loop to keep track of the consecutive errors + do a backoff accordingly (i.e. don't do steps by nilling the stepReqCh temporarily)

Reply:
Got it, sorry overlooked that we're returning `nil` unconditionally.

Reply:
Updated to handle errs in a buffered channel and added case in the event loop select to dequeue and backoff by setting `stepReqCh` to nil. We're _always_ returning the err in the backoff func now, so it will set `stepReqCh` to nil until _all_ the errs are dequeued and so we are backing off from more derivation steps exponentially.